shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Protocol.hs

```diff
@@ -77,6 +81,15 @@ import qualified Shelley.Spec.Ledger.STS.Tickn as STS.Tickn
 import Shelley.Spec.Ledger.Serialization (decodeRecordNamed)
 import Shelley.Spec.Ledger.Slot (SlotNo)
 
+class
+  ( CC.Crypto c,
+    DSignable c (OCertSignable c),
+    DSignable c (Hash c EraIndependentTxBody),
+    KESignable c (BHBody c),
+    VRFSignable c Seed
+  ) =>
+  TPraosCrypto c
```

Comment:
I'm open to suggestions for a better name and/or a better place for it.

Reply:
What does the `T` stand for?

Reply:
Transitional Praos (transition from PBFT to Praos). `PraosCrypto` would also be fine for me. Or `ShelleyCrypto`, or `ShelleyBasedCrypto`, ...

Reply:
ah! thanks :) I like `PraosCrypto` or `ShelleyCrypto`. I don't have strong opinions here.

Reply:
I think I prefer `ShelleyBasedCrypto` over `ShelleyCrypto` because the latter might give the impression that it is related to the Shelley era while it is in fact not specific to the Shelley era? :thinking: 

`PraosCrypto` is shorter and avoids the confusion?

Up to you.